### ISSUE TITLE
mt_a2a test failing

### ISSUE THREAD

**Issue by jdinan** [Sandia-OpenSHMEM/SOS#286]
The mt_a2a test is failing occasionally in Travis CI with the OFI transport with the following message:

```
Starting multithreaded test on 2 PEs, 8 threads/PE
Atomic test error: [PE = 1 | TID = 7] -- dest = -1 (expected 1), flag = 2 (expected 2)
Success
```

**Comment by jdinan** [comment]
Another failure message:
```
Starting multithreaded test on 2 PEs, 8 threads/PE
Atomic test error: [PE = 1 | TID = 7] -- dest = -1 (expected 1), flag = 2 (expected 2)
Atomic test error: [PE = 1 | TID = 5] -- dest = -1 (expected 1), flag = 2 (expected 2)
Atomic test error: [PE = 1 | TID = 3] -- dest = -1 (expected 1), flag = 2 (expected 2)
Atomic test error: [PE = 1 | TID = 4] -- dest = -1 (expected 1), flag = 2 (expected 2)
Atomic test error: [PE = 1 | TID = 2] -- dest = -1 (expected 1), flag = 2 (expected 2)
```

**Comment by jdinan** [comment]
This bug has not manifested since #352.

**Comment by jdinan** [comment]
Still failing in v1.3.2

**Comment by jdinan** [comment]
Fixed (finally?) by #418
